From 7a4f578a1a629c8a25c971ede0c190a02d83ae7b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Wed, 19 Mar 2014 16:03:14 +0100
Subject: [PATCH] Fix t/dhe.t for openssl 1.0.1beta
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This is port from 1.958 to 1.31 of these upstream commits:

Required since openssl-1.0.1e-39.el6:

commit ed5715eaf61c8cb54c2c06f95c2a328fd9f039cb
Author: Steffen Ullrich <Steffen_Ullrich@genua.de>
Date:   Sun Nov 10 16:26:25 2013 +0100

    - fix some tests

commit 00483ba8fe33cf0559bda8c20b892ab10aa48d41
Author: Steffen Ullrich <github@maulwuff.de>
Date:   Fri May 11 21:27:40 2012 +0200

    1.73 fixes to t/dhe.t to support more openssl versions

Required since openssl-1.0.1e-15.el6:

commit ef87a2b97c2d04714bd900496c20465c8c92b25a
Author: Steffen Ullrich <github@maulwuff.de>
Date:   Sun Feb 26 23:07:09 2012 +0100

    1.58 - disable workaround in  t/dhe.t for older openssl versions

commit 266ecce299e38dcd1f1d66223231bc6c86616b61
Author: Steffen Ullrich <github@maulwuff.de>
Date:   Sun Feb 26 22:57:06 2012 +0100

    1.57 - fix t/dhe.t for openssl 1.0.1beta

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 t/dhe.t | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/t/dhe.t b/t/dhe.t
index f50b6e6..b264a33 100644
--- a/t/dhe.t
+++ b/t/dhe.t
@@ -32,6 +32,10 @@ my $server = IO::Socket::SSL->new(
     SSL_cert_file => "certs/server-rsa384-dh.pem",
     SSL_key_file  => "certs/server-rsa384-dh.pem",
     SSL_dh_file   => "certs/server-rsa384-dh.pem",
+    # at least 0.9.8[ab] have problems if we don't explicitly disable
+    # RSA or EXPORT56, and 1.0.1 complains if we have RSA authentication
+    # enabled
+    SSL_cipher_list => 'ALL:RSA:!aRSA',
 ) || do {
     notok($!);
     exit
@@ -49,7 +53,9 @@ if ( !defined $pid ) {
 
     $ID = 'client';
     close($server);
-    my $to_server = IO::Socket::SSL->new( $addr ) || do {
+    my $to_server = IO::Socket::SSL->new(
+        PeerAddr => $addr,
+        SSL_cipher_list => 'ALL:RSA:!aRSA' ) || do {
     	notok( "connect failed: ".IO::Socket::SSL->errstr() );
 	exit
     };
-- 
2.4.3

